### ISSUE TITLE
az login with WAM fails due to 'Account has previously been signed out of this application'

### ISSUE THREAD

**Issue by bgrainger** [Azure/azure-cli#26160]
## Describe the bug

`az login` with the new Web Account Manager (WAM) broker fails to log in.

**Command Name**
`az login`

**Errors:**

```
Failed to authenticate [Tenant ID] '[Company Name]' due to error 'Account has previously been signed out of this application.. Status: Response_Status.Status_AccountUnusable, Error code: 0, Tag: 540940121'
```

## To Reproduce:

When I ran `az login`, the post-login screen told me:

> **Announcements**
> [Windows only] Starting in May 2023, Azure CLI will authenticate using the Web Account Manager (WAM) broker by default.
> 
> To help us collect feedback on the new login experience, you may opt-in to use WAM by running the following commands:
> 
> az config set core.allow_broker=true
> az account clear
> az login

I ran those three commands. After `az login`, a dialog (that seemed like native Windows UI) appeared and asked me to select an account. I selected my regular Work or School account. After I did that, the error message above was displayed, and I was not logged in.

## Expected Behavior

`az login` would succeed with my work account when using the WAM.

## Environment Summary
```
Windows-10-10.0.19045-SP0
Python 3.10.10
Installer: MSI

azure-cli 2.47.0

Extensions:
account 0.2.5
aks-preview 0.5.136
application-insights 0.1.16
azure-devops 0.25.0
containerapp 0.3.27
ssh 1.1.5

Dependencies:
msal 1.20.0
azure-mgmt-resource 22.0.0
```
## Additional Context

Also filed as Azure support request ID 2304120010002244.

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by bgrainger** [comment]
[login.txt](https://github.com/Azure/azure-cli/files/11254152/login.txt)

Redacted log attached. If you need the full details, I can email it to the address of your choice.

**Comment by Szeraax** [comment]
Same deal here. Decided to try out the WAM method. Got the same error. Gonna be a big pain if all our users have this happen in 2 weeks. Hopefully there is already a method to force the existing login method. 

Here's my log:

[login_szeraax.txt](https://github.com/Azure/azure-cli/files/11298330/login_szeraax.txt)

**Comment by mfergu98** [comment]
Hi All:  Having a similar issue after updating my az cli to version 2.48.1 ... I was able to get to the smaller work login screen that prompted the MS Authenticator MFA, but then failed with error below.  I have attached the redacted logs, if that helps to better reproduce and/or fix the issue.

cli.azure.cli.core._profile: Failed to authenticate {work_tenant} '{workplace}' due to error 'Account has previously been signed out of this application.. Status: Response_Status.Status_AccountUnusable, Error code: 0, Tag: 540940121'
{
  "azure-cli": "2.48.1",
  "azure-cli-core": "2.48.1",
  "azure-cli-telemetry": "1.0.8",
  "extensions": {
    "account": "0.2.5",
    "arcdata": "1.4.13",
    "logic": "0.1.6"
  }
}

[redacted_azlogin_failure_debug_logs.txt](https://github.com/Azure/azure-cli/files/11343036/redacted_azlogin_failure_debug_logs.txt)

**Comment by agravity-philipp** [comment]
On my side the authentication fails as well with: 
`Failed to authenticate <tenantID> '<tenantName>' due to error 'Account has previously been signed out of this application.. Status: Response_Status.Status_AccountUnusable, Error code: 0, Tag: 540940121'`

**Comment by jhew** [comment]
I was able to circumvent this issue by rerunning the command with allow_broker=false
az config set core.allow_broker=false
az account clear
az login

**Comment by agravity-philipp** [comment]
> I was able to circumvent this issue by rerunning the command with allow_broker=false az config set core.allow_broker=false az account clear az login

Me too! It seems that this does not use WAM anymore.

**Comment by rayluo** [comment]
We believe this has been fixed in the latest version of Azure CLI 2.49. Can we close this issue now, @jiasli ? CC: @MSamWils

**Comment by bgrainger** [comment]
Confirmed that `az login` is working for me (with the new broker) in v2.49.0/
